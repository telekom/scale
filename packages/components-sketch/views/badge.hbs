
<div data-sketch-artboard="Callout" class="artboard">
  <h1 class="title--artboard">Overview</h1>

  <h2 class="title--artboard-section">Callout Eyecatcher</h2>
  <div class="agent-states--grid" data-category="Text only">
    <div class="agent-states--item">
      <p class="agent-states--label">Primary - Large</p>
      <scale-callout size="large" color="primary" rotation="0">
      <div>
        <div
          style="
            text-align: center;
            font-size: 20px;
            color: white;
            font-weight: 400;
          "
        >
          Für nur
        </div>
        <div
          style="
            text-align: center;
            font-size: 60px;
            color: white;
            font-weight: 700;
          "
        >
          <span> 15 € </span>
        </div>
      </div>
    </scale-callout>
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">White - Large</p>
      <scale-callout size="large" color="white" rotation="0">
      <div>
        <div
          style="
            text-align: center;
            font-size: 20px;
            color: black;
            font-weight: 400;
          "
        >
          Für nur
        </div>
        <div
          style="
            text-align: center;
            font-size: 60px;
            color: black;
            font-weight: 700;
          "
        >
          <span> 15 € </span>
        </div>
      </div>
    </scale-callout>
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">Black - Large</p>
      <scale-callout size="large" color="black" rotation="0">
      <div>
        <div
          style="
            text-align: center;
            font-size: 20px;
            color: white;
            font-weight: 400;
          "
        >
          Für nur
        </div>
        <div
          style="
            text-align: center;
            font-size: 60px;
            color: white;
            font-weight: 700;
          "
        >
          <span> 15 € </span>
        </div>
      </div>
    </scale-callout>
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">Colored - Large</p>
      <scale-callout size="large" color="blue" rotation="0">
      <div>
        <div
          style="
            text-align: center;
            font-size: 20px;
            color: white;
            font-weight: 400;
          "
        >
          Für nur
        </div>
        <div
          style="
            text-align: center;
            font-size: 60px;
            color: white;
            font-weight: 700;
          "
        >
          <span> 15 € </span>
        </div>
      </div>
    </scale-callout>
    </div>
  </div>

  <h2 class="title--artboard-section"></h2>
  <div class="agent-states--grid" data-category="Text only">
    <div class="agent-states--item">
      <p class="agent-states--label">Primary - Small</p>
      <scale-callout size="small" color="primary" rotation="0">
      <div>
        <div
          style="
            text-align: center;
            font-size: 20px;
            color: white;
            font-weight: 400;
          "
        >
          Für nur
        </div>
        <div
          style="
            text-align: center;
            font-size: 60px;
            color: white;
            font-weight: 700;
          "
        >
          <span> 15 € </span>
        </div>
      </div>
    </scale-callout>
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">White - small</p>
      <scale-callout size="small" color="white" rotation="0">
      <div>
        <div
          style="
            text-align: center;
            font-size: 20px;
            color: black;
            font-weight: 400;
          "
        >
          Für nur
        </div>
        <div
          style="
            text-align: center;
            font-size: 60px;
            color: black;
            font-weight: 700;
          "
        >
          <span> 15 € </span>
        </div>
      </div>
    </scale-callout>
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">Black - Small</p>
      <scale-callout size="small" color="black" rotation="0">
      <div>
        <div
          style="
            text-align: center;
            font-size: 20px;
            color: white;
            font-weight: 400;
          "
        >
          Für nur
        </div>
        <div
          style="
            text-align: center;
            font-size: 60px;
            color: white;
            font-weight: 700;
          "
        >
          <span> 15 € </span>
        </div>
      </div>
    </scale-callout>
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">Colored - small</p>
      <scale-callout size="small" color="blue" rotation="0">
      <div>
        <div
          style="
            text-align: center;
            font-size: 20px;
            color: white;
            font-weight: 400;
          "
        >
          Für nur
        </div>
        <div
          style="
            text-align: center;
            font-size: 60px;
            color: white;
            font-weight: 700;
          "
        >
          <span> 15 € </span>
        </div>
      </div>
    </scale-callout>
    </div>
  </div>

  <h2 class="title--artboard-section"></h2>
  <div class="agent-states--grid" data-category="Text only">
    <div class="agent-states--item">
      <p class="agent-states--label">Rotation 0˚ Default</p>
      <scale-callout size="large" color="primary" rotation="0">
      <div>
        <div
          style="
            text-align: center;
            font-size: 20px;
            color: white;
            font-weight: 400;
          "
        > Für nur
        </div>
        <div
          style="
            text-align: center;
            font-size: 60px;
            color: white;
            font-weight: 700;
          "
        > <span> 15 € </span>
        </div>
      </div>
    </scale-callout>
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">Rotation 30˚ </p>
      <scale-callout size="large" color="blue" rotation="30">
      <div>
        <div
          style="
            text-align: center;
            font-size: 20px;
            color: black;
            font-weight: 400;
          "
        > Für nur
        </div>
        <div
          style="
            text-align: center;
            font-size: 60px;
            color: black;
            font-weight: 700;
          "
        > <span> 15 € </span>
        </div>
      </div>
    </scale-callout>
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">Rotation 90˚ </p>
      <scale-callout size="large" color="black" rotation="90">
      <div>
        <div
          style="
            text-align: center;
            font-size: 20px;
            color: white;
            font-weight: 400;
          "
        > Für nur
        </div>
        <div
          style="
            text-align: center;
            font-size: 60px;
            color: white;
            font-weight: 700;
          "
        >
        <span> 15 € </span>
        </div>
      </div>
    </scale-callout>
    </div>
  </div>
  </div>
</div>

<script src="/symbol_names.js"></script>
<script>
  Array.from(document.querySelectorAll('scale-callout')).forEach((btn, i) => {
    btn.dataset.sketchSymbol = `Callout / ${order(
      1,
      getCategoryName(btn)
    )} / ${order(
      2,
      btn.dataset.hasIcon ? 'With Icon' : 'Without Icon'
    )} / ${order(4, getStateName(btn))}`;
  });
</script>






























