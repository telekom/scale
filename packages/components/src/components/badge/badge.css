:host {
  --color-badge: var(--scl-color-primary);
  --inner-width-small: var(--scl-spacing-80);
  --inner-height-small: var(--scl-spacing-80);
  --inner-width-large: 126.5px;
  --inner-height-large: 96px;
  --width-small: 120px;
  --height-small: 120px;
  --height-large: 160px;
  --width-large: 160px;
  --blue-badge: var(--scl-color-blue-60);
  --white-badge: var(--scl-color-white);
  --black-badge: var(--scl-color-black);
  --primary-badge: var(--scl-color-primary);
  --font-content: var(--scl-font-family-sans);
}

.badge {
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 100%;
  background: var(--color-badge);
  font-family: var(--font-content);
}

.badge.badge--color-primary {
  background: var(--primary-badge);
}
.badge.badge--color-white {
  background: var(--white-badge);
}
.badge.badge--color-blue {
  background: var(--blue-badge);
}
.badge.badge--color-black {
  background: var(--black-badge);
}

.badge.badge--size-large {
  width: var(--width-large);
  height: var(--height-large);
}

.badge.badge--size-small {
  width: var(--width-small);
  height: var(--height-small);
}

.badge.badge--size-large .badge--inner {
  width: var(--inner-width-large);
  height: var(--inner-height-large);
}

.badge.badge--size-small .badge--inner {
  width: var(--inner-width-small);
  height: var(--inner-height-small);
}

.badge--inner {
  transform: rotateZ(var(--badge-rotation));
  overflow: hidden;
}
