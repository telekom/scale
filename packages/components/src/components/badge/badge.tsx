import { Component, h, Host, Prop, Element } from '@stencil/core';
import classNames from 'classnames';
import statusNote from '../../utils/status-note';

@Component({
  tag: 'scale-badge',
  styleUrl: 'badge.css',
  shadow: true,
})
export class Badge {
  @Element() hostElement: HTMLElement;
  /** (optional) Variant size of the badge itself */
  @Prop({ mutable: true }) size: 'large' | 'small' = 'large';
  /** (optional) Variant color/filling of the badge */
  @Prop({ mutable: true }) color: 'primary' | 'white' | 'black' | 'blue';
  /** (optional) Variant rotation of the badge/circle */
  @Prop({ mutable: true }) rotation: number = 0;

  connectedCallback() {
    statusNote({ source: this.hostElement, tag: 'beta' });
  }

  displayStyle() {
    return `:host {
      --badge-rotation: ${this.rotation}deg;
    }`;
  }

  render() {
    return (
      <Host>
        <style>{this.displayStyle()}</style>
        <div part={this.getBasePartMap()} class={this.getCssClassMap()}>
          <div part="inner" class="badge--inner">
            <slot></slot>
          </div>
        </div>
      </Host>
    );
  }

  getBasePartMap() {
    return this.getCssOrBasePartMap('basePart');
  }

  getCssClassMap() {
    return this.getCssOrBasePartMap('css');
  }

  getCssOrBasePartMap(mode: 'basePart' | 'css') {
    const name = 'badge';
    const prefix = mode === 'basePart' ? '' : `${name}--`;

    return classNames(
      name,
      this.color && `${prefix}color-${this.color}`,
      this.size && `${prefix}size-${this.size}`
    );
  }
}
