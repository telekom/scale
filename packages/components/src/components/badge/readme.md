# scale-badge



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description                                     | Type                                        | Default     |
| ---------- | ---------- | ----------------------------------------------- | ------------------------------------------- | ----------- |
| `color`    | `color`    | (optional) Variant color/filling of the badge   | `"black" \| "blue" \| "primary" \| "white"` | `undefined` |
| `rotation` | `rotation` | (optional) Variant rotation of the badge/circle | `number`                                    | `0`         |
| `size`     | `size`     | (optional) Variant size of the badge itself     | `"large" \| "small"`                        | `'large'`   |


## Shadow Parts

| Part      | Description |
| --------- | ----------- |
| `"inner"` |             |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
