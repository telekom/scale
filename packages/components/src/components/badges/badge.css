/**
 * @prop --badge-display: Display of the badge
 * @prop --badge-padding: Padding of the badge
 * @prop --badge-font-size: Font size of the badge
 * @prop --badge-font-weight: Font weight of the badge
 * @prop --badge-line-height: Line height of the badge
 * @prop --badge-text-align: Text align of the badge
 * @prop --badge-white-space: White space of the badge
 * @prop --badge-vertical-align: Vertical align of the badge
 * @prop --badge-border-radius: Border radius of the badge
 * @prop --badge-transition: Transition of the badge
 * @prop --badge-background-color: Background of the badge
 * @prop --badge-color: Color of the text of the badge
 * @prop --badge-pill-padding-right: Padding right of the pill of the badge
 * @prop --badge-pill-padding-left: Padding left of the pill of the badge
 * @prop --badge-pill-border-radius: Border radius of the pill of the badge
 * @prop --badge-pill-background-color: Background color of the pill of the badge
 * @prop --badge-pill-color: Color of the text of the pill of the badge
 * @prop --badge-link-text-decoration: Text decoration of the link of the badge
 * @prop --badge-link-hover-background: Hover background of the link of the badge
 * @prop --badge-link-active-background: Active background of the link of the badge
 * @prop --badge-link-focus-outline: Focus outline of the link of the badge
 * @prop --badge-link-focus-box-shadow: Box shadow of the link of the badge 
 */

.badge {
  display: var(--badge-display, inline-block);
  padding: var(--badge-padding, 0.25em 0.4em);
  font-size: var(--badge-font-size, 90%);
  font-weight: var(--badge-font-weight, 700);
  line-height: var(--badge-line-height, 1);
  text-align: var(--badge-text-align, center);
  white-space: var(--badge-white-space, nowrap);
  vertical-align: var(--badge-vertical-align, baseline);
  border-radius: var(--badge-border-radius, 0.25rem);
  transition: var(
    --badge-transition,
    color 0.15s ease-in-out,
    background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out,
    box-shadow 0.15s ease-in-out
  );
  background-color: var(--badge-background-color, #343a40);
  color: var(--badge-color, #fff);
}
.badge--pill {
  padding-right: var(--badge-pill-padding-right, 0.6em);
  padding-left: var(--badge-pill-padding-left, 0.6em);
  border-radius: var(--badge-pill-border-radius, 10rem);
  background-color: var(--badge-pill-background-color, #343a40);
  color: var(--badge-pill-color, #fff);
}

.badge--variant-dark {
  color: #fff;
  background-color: #343a40;
}

.badge--variant-light {
  color: #212529;
  background-color: #f8f9fa;
}

.badge--variant-info {
  color: #fff;
  background-color: #17a2b8;
}

.badge--variant-warning {
  color: #212529;
  background-color: #ffc107;
}

.badge--variant-danger {
  color: #fff;
  background-color: #dc3545;
}

.badge--variant-success {
  color: #fff;
  background-color: #28a745;
}

.badge--variant-secondary {
  color: #fff;
  background-color: #6c757d;
}

.badge--variant-primary {
  color: #fff;
  background-color: #007bff;
}
.badge--link {
  text-decoration: var(--badge-link-text-decoration, none);
}

.badge--link:hover {
  background: var(--badge-link-hover-background, #1d2124);
}

.badge--link:active {
  background: var(--badge-link-active-background, #1d2124);
}

.badge--link:focus {
  outline: var(--badge-link-focus-outline, 0);
  box-shadow: var(
    --badge-link-focus-box-shadow,
    0 0 0 0.2rem rgba(52, 58, 64, 0.5)
  );
}
