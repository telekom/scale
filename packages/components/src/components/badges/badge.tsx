import { Component, Prop, h } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';

@Component({
  tag: 't-badge',
  styleUrl: 'badge.css',
  shadow: true,
})
export class Badge {
  /** (optional) Badge class */
  @Prop() public customClass?: string = '';
  /** (optional) Badge size */
  @Prop() public size?: string = '';
  /** (optional) Badge variant */
  @Prop() public variant?: string = '';
  /** (optional) Badge pill */
  @Prop() public pill?: boolean = false;
  /** (optional) Badge link */
  @Prop() public link?: string = '';

  public render() {
    if (!!this.link) {
      return (
        <a href={this.link} class={this.getCssClassMap()}>
          <slot />
        </a>
      );
    }
    return (
      <span class={this.getCssClassMap()}>
        <slot />
      </span>
    );
  }

  private getCssClassMap(): CssClassMap {
    return classNames(
      'badge',
      this.customClass && this.customClass,
      this.size && `badge--size-${this.size}`,
      this.variant && `badge--variant-${this.variant}`,
      this.pill && `badge--pill`,
      !!this.link && 'badge--link'
    );
  }
}
