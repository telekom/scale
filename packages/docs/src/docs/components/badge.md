---
title: badge
description: badge element
url: /docs/badge
contributors:
  - SylviaGonschior
---

# Badge

Badge component

selector: `t-badge`

## Default

```javascript
<t-badge>default</t-badge>
```

<div class="demo-container">
  <t-badge>default</t-badge>
</div>

## Pill

```javascript
<t-badge pill="true">Pill</t-badge>
```

<div class="demo-container">
 <t-badge pill="true">Pill</t-badge>
</div>

## With link

```javascript
<t-badge link="#">badge with link</t-badge>
```

<div class="demo-container">
 <t-badge link="#">badge with link</t-badge>
</div>

## Variants

Available `variants`:

- dark
- light
- info
- warning
- danger
- success
- primary
- secondary

```javascript
<t-badge variant="primary">Primary</t-badge>
```

<div class="demo-container">
 <t-badge variant="primary">Primary</t-badge>
</div>
