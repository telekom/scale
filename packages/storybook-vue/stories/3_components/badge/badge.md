<div style="display: inline-flex; align-items: center; justify-content: space-between; width: 100%;">
    <h1>Badge</h1>
    <img src="assets/aa.png" alt="Accessible AA" />
</div>

Use the Badge Component  to draw attention to another interface element or to display a notification. It enhances the component to which it is attached with additional information, disclosed to the user upon interacting with it.

## General

The Badge can carry different types of content such as a letters,numbers or icons.

IMAGE EXAMPLES

### When to use

Use a badge in the following cases:

- When items need to be differentiated or become an eyecatcher.

### When not to use

Don't use cards:

- If lables are too long

## Variants
## Color

### Primary / White / Black / Colored

Primary is the default color but the user also has the possibility to determine any other color as well.
IMAGE EXAMPLES


## Sizes
There is two types: large & small

### Large Badge

The large badge is the defualt size. It has a size of 160x160 and the content area a size of 96x126.5

IMAGE

### Small Badge

The small badge has a size of 120x120 and the content area a size of 88x86.

IMAGE


## Rotation

The standard is a 0 degrees rotation but it the user has the possibility to determine.
For example a 30 degrees rotation:

IMAGE

### Best practice 

## Content
- if possible, try not to use more than three words.
- Only the first word of the label should be capitalized.

## Background
- The background color can be used to emphasize a status, to inform or to associate information.

## Behavior
- The badge component has no interaction. It is not clickable or removable. 

## Design on your own
- Choose the right size.
- Choose the right color.
- Choose a rotation if needed.
- Write your content (watch out for "When not to use").



