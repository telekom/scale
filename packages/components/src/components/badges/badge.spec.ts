import { newSpecPage } from '@stencil/core/testing';
import { Badge } from './badge';

describe('Badge', () => {
  let element;
  beforeEach(async () => {
    element = new Badge();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Badge],
      html: `<t-badge>Label</t-badge>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should have a default css class', () => {
    expect(element.getCssClassMap()).toBe('badge');
  });

  it('should have a link', async () => {
    const page = await newSpecPage({
      components: [Badge],
      html: `<t-badge link="#">Label</t-badge>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle size css class', () => {
    element.size = 'small';
    expect(element.getCssClassMap()).toContain('badge--size-small');
  });

  it('should have class variant primary', () => {
    element.variant = 'primary';
    expect(element.getCssClassMap()).toContain('badge--variant-primary');
  });

  it('should render pill badge', () => {
    element.pill = true;
    expect(element.getCssClassMap()).toContain('badge--pill');
  });
});
