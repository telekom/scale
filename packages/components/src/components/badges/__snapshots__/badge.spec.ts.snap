// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should have a link 1`] = `
<t-badge link="#">
  <mock:shadow-root>
    <a class="badge badge--link" href="#">
      <slot></slot>
    </a>
  </mock:shadow-root>
  Label
</t-badge>
`;

exports[`Badge should match snapshot 1`] = `
<t-badge>
  <mock:shadow-root>
    <span class="badge">
      <slot></slot>
    </span>
  </mock:shadow-root>
  Label
</t-badge>
`;
